mongodb/trigger/mongodb/trigger.go

```diff
@@ -0,0 +1,248 @@
+			eventOption = 3
+			pipeline = mongo.Pipeline{bson.D{{"$match",
+				bson.D{{"operationType", "delete"}},
+			}}}
+		} else if evntLsnr.settings.ListenInsert && evntLsnr.settings.ListenUpdate && !evntLsnr.settings.ListenRemove {
+			eventOption = 4
+			pipeline = mongo.Pipeline{bson.D{{"$match", bson.D{{"$or",
+				bson.A{
+					bson.D{{"operationType", "insert"}},
+					bson.D{{"operationType", "update"}}}}},
+			}}}
+		} else if evntLsnr.settings.ListenInsert && evntLsnr.settings.ListenRemove && !evntLsnr.settings.ListenUpdate {
+			eventOption = 5
+			pipeline = mongo.Pipeline{bson.D{{"$match", bson.D{{"$or",
+				bson.A{
+					bson.D{{"operationType", "insert"}},
+					bson.D{{"operationType", "delete"}}}}},
+			}}}
+		} else if evntLsnr.settings.ListenRemove && evntLsnr.settings.ListenUpdate && !evntLsnr.settings.ListenInsert {
+			eventOption = 6
+			pipeline = mongo.Pipeline{bson.D{{"$match", bson.D{{"$or",
+				bson.A{
+					bson.D{{"operationType", "delete"}},
+					bson.D{{"operationType", "update"}}}}},
+			}}}
+		} else {
+			pipeline = mongo.Pipeline{}
+		}
+		fmt.Println("====eventOption=== %d", eventOption)
+		t.logger.Debugf("====eventOption=== %d", eventOption)
+		db := t.mclient.Database(evntLsnr.database)
+		var stream *mongo.ChangeStream
+		var err error
+		if evntLsnr.settings.Collection != "" {
+			coll := db.Collection(evntLsnr.settings.Collection)
+			t.logger.Infof("listening on collection:: %s", evntLsnr.settings.Collection)
+			stream, err = coll.Watch(context.Background(), pipeline)
+		} else { // listening on database if no collection name specified
+			t.logger.Infof("listening on all collections of database:: %s", evntLsnr.database)
+			stream, err = db.Watch(context.Background(), pipeline)
+		}
+
+		if err != nil {
+			t.logger.Errorf("Failed to open change stream %s", err)
+			return err
+		}
+		// Start polling on a separate Go routine so as to not block engine
+		go evntLsnr.listen(stream)
+	}
+	t.logger.Infof("Trigger - %s  started", t.id)
+	return nil
+}
+
+func (entLsnr *EventListener) listen(stream *mongo.ChangeStream) {
+	entLsnr.logger.Infof("============listening====")
+	for {
+		select {
+		case <-entLsnr.done:
+			entLsnr.logger.Infof("stopped listening...")
+			// Exit
+			return
+		default:
+			//	entLsnr.logger.Infof("done in listening %b", entLsnr.done)
+			ok := stream.Next(context.Background())
+			if ok {
+				var res bson.D
+				err := stream.Decode(&res)
+				if err != nil {
+					entLsnr.logger.Errorf("got error while decoding stream %s", err)
+				}
+				if len(res) == 0 {
+					entLsnr.logger.Infof("result is empty, was expecting change document")
+				}
+				stringOp := stream.Current.String()
+				go entLsnr.process(stringOp, entLsnr)
+			} else {
+				err := stream.Err()
+				if err != nil {
+					//if err is not nil, it means something bad happened, let's finish our func
+					stream.Close(context.Background())
+					entLsnr.logger.Infof("stopped listening...stream closed")
```

Comment:
Are you not supposed to show the error here ?

Reply:
will show the error in error level log.